src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -1637,6 +1822,163 @@ subroutine wrfv3_netcdf(fv3filenamegin)
        call gsi_fv3ncdf_writeps_v1(dynvars,'ps',ges_ps,mype_p,add_saved)
     
     endif
+
+! additional I/O for direct reflectivity DA capabilities
+    if (l_use_dbz_directDA) then
+! Initialized temp arrays with ges. Will be recalculated later if log(q) is used
+       tmparr_qr =ges_qr
+       tmparr_qs =ges_qs
+       tmparr_qg =ges_qg
+       tmparr_qnr=ges_qnr
+
+       do k=1,nsig
+         do i=1,lon2
+           do j=1,lat2
+
+!             initialize hydrometeors as zero
```

Comment:
Just like my previous comment on the similarity of the code. Try to use a subroutine to reduce the duplication of the code.

Reply:
I created to subroutines according to your comment since these codes also exist in WRF code. I replaced these parts with the subroutines in ‘gsi_rfv3io_mod.f90’ and ‘cplr_read_wrf_mass_guess.f90’ and ‘cplr_wrwrfmassa.f90’.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-4c910ebd862ea8bea4024bbfa73835e643efb4091155d4e8ef121388b14ab6ffR2744-R3211